libdevcore/CommonData.h

```diff
@@ -47,6 +49,32 @@ enum class HexPrefix
 	DontAdd = 0,
 	Add = 1,
 };
+
+/// Joins collection of strings into one string with separators between, last separator can be different.
+/// @param _list collection of strings to join
+/// @param _separator defaults to ", "
+/// @param _lastSeparator (optional) will be used to separate last two strings instead of _separator
+/// @example join(vector<string>{"a", "b", "c"}, "; ", " or ") == "a; b or c"
+template<class T>
+std::string join(
+	const T& _list,
+	const std::string& _separator =", ",
+	const std::string& _lastSeparator = ""
+) {
+	std::string result = boost::algorithm::join(_list, _separator);
+
+	// calling .back() on empty _list is undefined behavior
+	if (!_lastSeparator.empty() && !_list.empty()) {
```

Comment:
Should this allow empty lists as input?

If yes, should we just have a `if (_list.empty()) return string{}` as the first statement before calling `join`?